engineapi/instance_manager.go

```diff
@@ -336,6 +336,10 @@ func getBinaryAndArgsForEngineProcessCreation(e *longhorn.Engine,
 		}
 	}
 
+	if engineCLIAPIVersion >= 9 {
+		args = append([]string{"--engine-instance-name", e.Name}, args...)
+	}
+
```

Comment:
Looks like we are adding argument to tail -> then top -> then tail which creates a tiny reading issue. WDYT?

Reply:
At first I was looking for a code correctness issue, but I think I understand you mean it makes the code harder to read.

IMO it is better to maintain the progression of engineCLIAPIVersion (e.g., first we add arguments assocaited with version 6, then version 7, then version 9). It's a personal preference that I don't feel too strongly about, so if you still disagree after reading this comment, let me know and I'll move the global flag addition higher up in the code.

Reply:
Sounds good to me to follow the progression of engineCLIAPIVersion. Let's keep this current implementation